code/modules/mob/living/carbon/human/species_types/mothmen.dm

```diff
@@ -47,3 +55,125 @@
+	if(H.has_status_effect(STATUS_EFFECT_COCOONED))
+		return TRUE //Cocooned mobs dont get to wake up
+*/
+/datum/action/innate/cocoon
+	name = "Cocoon"
+	desc = "Restore your wings and antennae, and heal some damage. If your cocoon is broken externally you will take heavy damage!"
+	check_flags = AB_CHECK_RESTRAINED|AB_CHECK_STUN|AB_CHECK_CONSCIOUS
+	button_icon_state = "wrap_0"
+	icon_icon = 'icons/mob/actions/actions_animal.dmi'
+
+//
+/datum/action/innate/cocoon/Activate()
+	var/mob/living/carbon/H = owner
+	var/obj/item/organ/wingcheck = H.getorgan(/obj/item/organ/wings/moth)
+	if(!wingcheck) //This is to stop easy organ farms
+		to_chat(H, "<span class='warning'>You don't have any wings to regenerate!</span>")
+		return
+	if(!(H.dna.features["moth_wings"] == "Burnt Off"))
+		to_chat(H, "<span class='warning'>Your wings are fine as they are!</span>")
+		return
+	if(H.nutrition < COCOON_NUTRITION_AMOUNT)
+		to_chat(H, "<span class='warning'>You are too hungry to weave a cocoon!</span>")
+		return
+	H.visible_message("<span class='notice'>[H] begins to hold still and concentrate on weaving a cocoon...</span>", \
+	"<span class='notice'>You begin to focus on weaving a cocoon... (This will take [COCOON_WEAVE_DELAY / 10] seconds and you must hold still.)</span>")
+	H.adjustStaminaLoss(20, 0) //this is here to deter people from spamming it if they get interrupted
+	if(do_after(H, COCOON_WEAVE_DELAY, FALSE, H))
+		if(!ismoth(H))
+			to_chat(H, "<span class='warning'>You have lost your mandibles and cannot weave anymore!.</span>")
+			return
+		if(H.incapacitated())
+			to_chat(H, "<span class='warning'>You cannot weave a cocoon in your current state.</span>")
+			return
+		H.visible_message("<span class='notice'>[H] finishes weaving a cocoon!</span>", "<span class='notice'>You finish weaving your cocoon.</span>")
+		var/obj/structure/moth/cocoon/C = new(get_turf(H))
+		H.forceMove(C)
+		//H.Sleeping(COCOON_EMERGE_DELAY, 0)
+		H.Sleeping(20, 0)
+		C.preparing_to_emerge = TRUE
+		H.apply_status_effect(STATUS_EFFECT_COCOONED)
+		//owner.log_message("[key_name(owner)] has finished weaving a cocoon at [AREACOORD(owner)]")
+		addtimer(CALLBACK(src, .proc/emerge, C), COCOON_EMERGE_DELAY, TIMER_UNIQUE)
+	else
+		to_chat(H, "<span class='warning'>You need to hold still in order to weave a cocoon!</span>")
+
+//Removes moth from cocoon, restores burnt wings
+/datum/action/innate/cocoon/proc/emerge(obj/structure/moth/cocoon/C)
+	for(var/mob/living/carbon/human/H in C.contents)
+		if(!H.has_status_effect(STATUS_EFFECT_COCOONED))
+			return
+		//if(H.dna.features["moth_wings"] == "Burnt Off") //this check seems redundant as burned wings are a roundstart option and sending a signal to clear a non-existing mood event doesn't cause any issues
+		SEND_SIGNAL(H, COMSIG_CLEAR_MOOD_EVENT, "burnt_wings")
+		//if(!ismoth(H))//mutation toxin is a thing, so this is here to prevent accidentally creating a mishmash of moth and whatever else
```

Comment:
if(!ismoth(H)
    break
will do what you want

Reply:
I tried that but then it straight up stopped me from waking up ever. So I'll just remove the commented parts entirely.